### PR TITLE
Update template to use the latest JSON format

### PR DESCRIPTION
**Pull request by theelderwand**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: create-budget/azuredeploy.json [hunk 1]
```diff
@@ -16,17 +16,6 @@
         "description": "The total amount of cost or usage to track with the budget"
       }
     },
-    "budgetCategory": {
```
Comment (bmoore-msft) [inline create-budget/azuredeploy.json:19]: Why are we removing this?

File: create-budget/azuredeploy.json [hunk 2]
```diff
@@ -51,21 +40,16 @@
         "description": "The end date for the budget in YYYY-MM-DD format. If not provided, we default this to 10 years from the start date."
       }
     },
-    "operator": {
```
Comment (bmoore-msft) [inline create-budget/azuredeploy.json:54]: Why are we removing this?

File: create-budget/azuredeploy.json [hunk 3]
```diff
@@ -76,71 +60,76 @@
         "description": "The list of email addresses to send the budget notification to when the threshold is exceeded."
       }
     },
-    "contactRoles": {
```
Comment (bmoore-msft) [inline create-budget/azuredeploy.json:79]: Why are we removing this?

File: create-budget/azuredeploy.parameters.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+  "$schema": "https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json#",
```
Comment (bmoore-msft) [inline create-budget/azuredeploy.parameters.json:2]: revert this change

File: create-budget/metadata.json [hunk 5]
```diff
@@ -2,12 +2,12 @@
   "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
   "type": "QuickStart",
   "itemDisplayName": "Create a Budget",
-  "description": "This template shows how to create a budget under a resource group. This feature is available to enterprise customers only",
-  "summary": "This template creates a budget to track cost/usage and get notified whenever a specified threshold is met. This feature is available to enterprise customers only",
+  "description": "This template shows how to create a budget under asubscription.",
```
Comment (bmoore-msft) [inline create-budget/metadata.json:5]: fix typo

File: create-budget/azuredeploy.json (hunk 2 above)
Comment (theelderwand) [reply create-budget/azuredeploy.json:54]: This has been made implicit in the ARM template.

File: create-budget/azuredeploy.json (hunk 1 above)
Comment (theelderwand) [reply create-budget/azuredeploy.json:19]: To simplify the budget document as a default is already set by the budget service. Let me know if you feel strongly about this, i can add it back.

File: create-budget/azuredeploy.json (hunk 3 above)
Comment (theelderwand) [reply create-budget/azuredeploy.json:79]: Added this one back.

File: create-budget/azuredeploy.parameters.json (hunk 4 above)
Comment (theelderwand) [reply create-budget/azuredeploy.parameters.json:2]: Why ? I'm intending this to be a budget created at the subscription level.

File: create-budget/metadata.json (hunk 5 above)
Comment (theelderwand) [reply create-budget/metadata.json:5]: fixed.

File: create-budget/azuredeploy.parameters.json (hunk 4 above)
Comment (bmoore-msft) [reply create-budget/azuredeploy.parameters.json:2]: this is not a valid schema for a parameters file

File: create-budget/azuredeploy.json (hunk 1 above)
Comment (bmoore-msft) [reply create-budget/azuredeploy.json:19]: seems like it makes the template more broadly applicable if the resource has the capability

File: create-budget/azuredeploy.parameters.json (hunk 4 above)
Comment (theelderwand) [reply create-budget/azuredeploy.parameters.json:2]: Ah! never mind. reverting.
